### DISCUSSION TITLE
Release 1.6.4

### DISCUSSION THREAD

**Discussion by vomikan** [moneymanagerex/moneymanagerex#5777] [General]
https://github.com/moneymanagerex/moneymanagerex/milestone/41?closed=1

A lot of work has been done. It's time to make a release.
Words of gratitude are all directed @n-stein 
Thank you so much for your contribution to the development of the product!

https://crowdin.com/project/moneymanagerex
Translations are still a little behind. Let's wait a week. It would be desirable to draw the attention of translators to this.

**Comment by n-stein** [comment]
I think these can all be closed, or at least marked as fixed/solution found if you want to keep them open for some reason:

#3148
#5451
#5568
#5676
#5683

**Comment by comradekingu** [comment]
Happy to help with the translations if you use Weblate. :)

**Reply by vomikan** [reply]
https://crowdin.com/project/moneymanagerex

**Reply by comradekingu** [reply]
I can't log in there without signing away rights to all my work.
As a tool, it lacks the ability to navigate large spans of strings at once, or see more languages at the same time, etc. etc.
https://hosted.weblate.org/projects/i-hate-money/ and https://hosted.weblate.org/projects/gnucash/
are already on the hosted variant.

**Reply by vomikan** [reply]
For my part, I don't see any drawbacks with this platform.
Volunteers can translate into any languages. Texts are automatically synchronized with a repository. Everything is perfect.

**Reply by comradekingu** [reply]
Save for the tool itself, and what happens in addition to that, without any care for what that text is, I agree.

**Comment by vomikan** [comment]
This release already contains 60 fixes. At startup, the program reads the page with[ releases](https://api.github.com/repos/moneymanagerex/moneymanagerex/releases). Each time it works slower and slower. I suggest removing the detailed list of bugs and leaving a link to the closed bugs.

Just this:
https://github.com/moneymanagerex/moneymanagerex/milestone/41?closed=1

But not this:
**Features/Enhancements:**

**Fixes:**
#5464	Improve budget perfomance report
#5571	Deleted Transaction Dialog Titlebar
#5573	Budget Performance Report
#5602	QIF import date format
#5606	Search Feature issue
#5614	Recurring transactions
#5615	Keyboard shortcuts
#5590 #5625 #5626 #5628 #5655 #5636 #5659 	Translation improvements
#5629	Currency dialog improvements
#5641	Imported transactions from webapp not working correctly
#5650	Add ellipsis to "View with Transaction Filter" menu option
#5658	Changes keyboard shortcuts in Accounts menu
#5670	MMEX crashes on asset transaction
#5686	Budget view crash on double click category summary

**Comment by kevyndev** [comment]
Hi guys 

there is already a date for 1.6.4?

**Reply by vomikan** [reply]
There is no specific date. You can focus on the percentage before the milestone is completed.

https://github.com/moneymanagerex/moneymanagerex/milestone/41

Actually, there is a date, but we are not obliged to observe the timing.

**Reply by kevyndev** [reply]
got it

thanks for the reply

**Comment by n-stein** [comment]
Are there any other particularly egregious bugs that need to be squashed for 1.6.4? Stock price update was definitely the major one.

There are no pending issues under the 1.6.4 milestone (current three open items are all fixed), but wanted to make sure there aren't any missing that should be included.

Also, please add #5759 to the 1.6.4 milestone. It's already closed, but I feel it should be included in the list of notable enhancements for this release.

**Comment by tactilis** [comment]
> Are there any other particularly egregious bugs that need to be squashed for 1.6.4?

I have just encountered this when upgrading a 1.6.3 (DB v17) .mmb to DB v18 using 1.6.4 Beta 6.

![image](https://github.com/moneymanagerex/moneymanagerex/assets/4060092/cc297213-ffb6-4992-a92f-b09d2a4fc2f6)

It's weird because it only happens sometimes. I'm trying to pin down a repeatable set of steps so you can investigate it.

What does INFOID: 8 relate to please? Knowing that might give me a clue.

**Reply by n-stein** [reply]
INFOID = 8 is just the index of ISUSED in INFOTABLE_V1.

It's the flag that MMEX uses to identify if the database is already open in another instance of MMEX. The ISUSED field is only set in one place during database shutdown:

https://github.com/moneymanagerex/moneymanagerex/blob/master/src/mmframe.cpp#L371-L372

Maybe the database you are upgrading is marked as locked by another MMEX instance? Try forcefully shutting down MMEX (with task manager for example), then upgrading the database.

**Reply by tactilis** [reply]
> Maybe the database you are upgrading is marked as locked by another MMEX instance? Try forcefully shutting down MMEX (with task manager for example), then upgrading the database.

It's not that. 
I've found a repeatable case. Just writing it up in a new issue...

**Reply by tactilis** [reply]
See issue https://github.com/moneymanagerex/moneymanagerex/issues/5964

**Comment by tactilis** [comment]
@n-stein
> Are there any other particularly egregious bugs that need to be squashed for 1.6.4?

I've just found this one https://github.com/moneymanagerex/moneymanagerex/issues/5966

I think it should be addressed for 1.6.4

**Reply by n-stein** [reply]
Thanks, that one was due to a change I made recently in #5857. Fixed now.

**Comment by vomikan** [comment]
https://github.com/moneymanagerex/moneymanagerex/milestone/41

I would do a release this or next week. We are being held by two bugs. One of which can be postponed.

**Reply by n-stein** [reply]
@whalley mentioned that #5988 isn't an issue when he builds with wxWidgets 3.2.1, so perhaps this one is not critical for the 1.6.4 release?

**Reply by whalley** [reply]
Yes, MacOS all good with wx3.2.1

**Reply by n-stein** [reply]
That leaves #1254 which looks like it will be closed by #6021 and will resolve all open items in 1.6.4 milestone.

**Comment by vomikan** [comment]
What about release right now?

**Reply by whalley** [reply]
Update to website to reference new release will need to wait until later today.

**Reply by whalley** [reply]
Update to website to reference new release will need to wait until later today.

**Reply by whalley** [reply]
Done.
